tests/vcs/git/GitCommitBearTest.py

```diff
@@ -298,16 +298,106 @@ def test_check_issue_reference(self):
+
+        self.git_commit('Shortlog\n\n'
+                        'First line, blablablablablabla.\n'
+                        'Another line, blablablablablabla.\n'
+                        'Resolves https://bitbucket.org/user/repo/issues/1/')
+        self.assertEqual(self.run_uut(
+                             body_close_issue=True),
+                         ['Invalid issue reference: '
+                          'https://bitbucket.org/user/repo/issues/1/'])
+
+        self.git_commit('Shortlog\n\n'
+                        'First line, blablablablablabla.\n'
+                        'Another line, blablablablablabla.\n'
+                        'Resolves https://bitbucket.org/user/repo/issues/1/')
+        self.assertEqual(self.run_uut(
+                             body_close_issue=True,
+                             body_close_issue_full_url=True),
+                         ['Host bitbucket does not support full issue '
+                          'reference.'])
+
+        # Adding BitBucket's ssh remote for testing
+        self.run_git_command('remote', 'set-url', 'test',
+                             'git@bitbucket.org:user/repo.git')
+
+        # Unsupported Host and Issue reference combination - Bitbucket
+        self.git_commit('Shortlog\n\n'
+                        'First line, blablablablablabla.\n'
+                        'Another line, blablablablablabla.\n'
+                        'Closes #1112')
+        self.assertEqual(self.run_uut(
+                             body_close_issue=True,
+                             body_close_issue_full_url=True),
+                         ['Host bitbucket does not support full issue '
+                          'reference.'])
+
+        self.git_commit('Shortlog\n\n'
+                        'First line, blablablablablabla.\n'
+                        'Another line, blablablablablabla.\n'
+                        'Closes #1112')
+        self.assertEqual(self.run_uut(
+                             body_close_issue=True), [])
+
+        self.git_commit('Shortlog\n\n'
+                        'First line, blablablablablabla.\n'
+                        'Another line, blablablablablabla.\n'
+                        'Fixes issue #1112')
+        self.assertEqual(self.run_uut(
+                             body_close_issue=True), [])
+
+        self.git_commit('Shortlog\n\n'
+                        'First line, blablablablablabla.\n'
+                        'Another line, blablablablablabla.\n'
+                        'Resolving    bug #1112')
+        self.assertEqual(self.run_uut(
+                             body_close_issue=True), [])
+
+        self.git_commit('Shortlog\n\n'
+                        'First line, blablablablablabla.\n'
+                        'Another line, blablablablablabla.\n'
+                        'Closesticket #1112')
+        self.assertEqual(self.run_uut(
+                             body_close_issue=True), [])
```

Comment:
this should be an error, yes?

Reply:
[This](https://bitbucket.org/virresh/issueresolvinggit/commits/3fd3a151d17fc77d93fce59b9a776ea48ccf2e2b) works as per guidelines :sweat_smile:

Reply:
I do not think anyone would have a reason to write messages in such a skewed way. Maybe we should have an option for this bear indicating whether to stick to core guidelines or readability guidelines.